rook/ensure.go

```diff
@@ -104,7 +104,12 @@ func (pcx *PackageContext) EnsurePackage(meta versioning.DependencyMeta, forceUp
 		print.Verb(meta, "need to clone new copy from cache")
 		repo, err = pcx.EnsureDependencyFromCache(meta, dependencyPath, false)
 		if err != nil {
-			return errors.Wrap(err, "failed to ensure dependency from cache")
+			errors.Wrap(err, "failed to ensure dependency from cache")
+			err2 := os.RemoveAll(dependencyPath)
+			if err2 != nil {
+				return errors.Wrap(err, "failed to remove corrupted dependency repo")
+			}
+			return
```

Comment:
It's worth doing some kind of retry logic here, just so the user doesn't have to run it twice. Mainly because the user has no idea that this error is non-idempotent and may behave differently on a second run.